llama-index-core/llama_index/core/vector_stores/simple.py

```diff
@@ -63,8 +63,16 @@ def filter_fn(node_id: str) -> bool:
             if metadata_value is None:
                 filter_matches = False
             elif isinstance(metadata_value, list):
-                if filter_.value not in metadata_value:
-                    filter_matches = False
+                if filter_.operator == "any":
```

Comment:
If we are going to move away from legacy filters, let's just implement all the filters? (sorry for the delay here 😓)